extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/WrongAnnotationUsageProcessor.java

```diff
@@ -38,24 +37,25 @@ void detect(ArcConfig config, ApplicationIndexBuildItem applicationIndex, Custom
 
         // Detect unsupported annotations
         List<UnsupportedAnnotation> unsupported = new ArrayList<>();
-        Function<AnnotationInstance, String> singletonFun = new Function<AnnotationInstance, String>() {
 
-            @Override
-            public String apply(AnnotationInstance annotationInstance) {
-                return String.format("%s declared on %s, use @jakarta.inject.Singleton instead",
-                        annotationInstance.toString(false), getTargetInfo(annotationInstance));
-            }
-        };
-        unsupported.add(new UnsupportedAnnotation(DotName.createSimple("com.google.inject.Singleton"), singletonFun));
-        unsupported.add(new UnsupportedAnnotation(DotName.createSimple("jakarta.ejb.Singleton"), singletonFun));
-        unsupported.add(new UnsupportedAnnotation(DotName.createSimple("groovy.lang.Singleton"), singletonFun));
-        unsupported.add(new UnsupportedAnnotation(DotName.createSimple("jakarta.ejb.Singleton"), singletonFun));
+        String correctSingleton = "@jakarta.inject.Singleton";
+        unsupported.add(new UnsupportedAnnotation("com.google.inject.Singleton", correctSingleton));
+        unsupported.add(new UnsupportedAnnotation("jakarta.ejb.Singleton", correctSingleton));
+        unsupported.add(new UnsupportedAnnotation("groovy.lang.Singleton", correctSingleton));
+
+        String correctInject = "@jakarta.inject.Inject";
+        unsupported.add(new UnsupportedAnnotation("javax.inject.Inject", correctInject));
+        unsupported.add(new UnsupportedAnnotation("com.google.inject.Inject", correctInject));
+        unsupported.add(new UnsupportedAnnotation("com.oracle.svm.core.annotate.Inject", correctInject));
+        unsupported.add(new UnsupportedAnnotation("org.gradle.internal.impldep.javax.inject.Inject",
+                correctInject));
```

Comment:
`@Inject` and `@Singleton` are the most common ones and the problematic cases are mainly when you migrate to Quarkus 3 - the compiler won't tell you about these in tests (as seen from changed files).
I am not aware of similar issue with `@ApplicationScoped`, at least not in basic dependency setup?

Reply:
`@ApplicationScoped` is CDI specific, so it's unlikely to be forked that much. I agree `@Inject` and `@Singleton` are the most important. We could add `@Named` as well. I'm not sure about `@Qualifier` and `@Scope`.

Reply:
Well, no reason really... except that if you use a wrong scope the bean class is just ignored and usually results in a build failure (because of an unsatisfied dependency) while if you use a wrong `@Inject` you can get  NPEs at runtime.

Reply:
Yeah OK. You will have `javax.inject` around in our test scope so I suppose it makes sense to limit ourselves to these.

I was more thinking of someone who just upgraded the Quarkus version without changing the source code but they will have a ton of other issues anyway so I suppose we can ignore `@ApplicationScoped`.

Reply:
We can try to address scopes in a separate pull request...